prediction_market_agent/agents/prophet_agent/deploy.py

```diff
@@ -14,22 +13,73 @@
 from prediction_market_agent_tooling.tools.betting_strategies.stretch_bet_between import (
     stretch_bet_between,
 )
+from prediction_market_agent_tooling.tools.langfuse_ import observe
 from prediction_market_agent_tooling.tools.utils import (
     prob_uncertainty,
     should_not_happen,
 )
 from prediction_prophet.benchmark.agents import (
     EmbeddingModel,
     OlasAgent,
+    Prediction,
     PredictionProphetAgent,
+    Research,
 )
 
 from prediction_market_agent.utils import DEFAULT_OPENAI_MODEL
 
 
+class PredictionProphetAgentObserved(PredictionProphetAgent):
```

Comment:
Subclasses Prophet's agents just to `@observe` the methods. I thought it doesn't make sense to force `@observe` in Prophet's repository.